scripts/main/albums.js

```diff
@@ -78,69 +82,67 @@ albums.load = function () {
 	}
 };
 
+/**
+ * @param {(Album|TagAlbum|SmartAlbum)} album
+ * @returns {void}
+ */
 albums.parse = function (album) {
 	if (!album.thumb) {
-		album.thumb = {};
-		album.thumb.id = "";
-		album.thumb.thumb = album.password === "1" ? "img/password.svg" : "img/no_images.svg";
-		album.thumb.type = "";
-		album.thumb.thumb2x = "";
+		album.thumb = {
+			id: "",
+			thumb: album.has_password ? "img/password.svg" : "img/no_images.svg",
+			type: "image/svg+xml",
+			thumb2x: null,
+		};
 	}
 };
 
-// TODO: REFACTOR THIS
+/**
+ * Normalizes the built-in smart albums.
+ *
+ * TODO: REFACTOR THIS
+ * TODO @ildyria: What did you intended to express by the TODO above? What did you want to be re-factored?
```

Comment:
that the code was unnecessary having a case analysis that should not be there. 
Those info should come directly from the server.

Reply:
I don't understand what you mean. What is "that data"? The code basically replaces the title of the built-in smart albums by localized names. As far as I can tell, most (maybe even all) localization is done on the front-end. So what info do mean which should come directly from the server?

Reply:
The title should be provided directly from the server, not have to do this case analysis to localize the title.
All the tools for the localization are available serverside, there is no reason for this. :)